homeassistant/components/deconz/gateway.py

```diff
@@ -111,15 +117,22 @@ def option_allow_new_devices(self) -> bool:
     # Callbacks
 
     @callback
-    def async_connection_status_callback(self, available) -> None:
+    def async_connection_status_callback(self, available: bool) -> None:
         """Handle signals of gateway connection status."""
         self.available = available
         self.ignore_state_updates = False
         async_dispatcher_send(self.hass, self.signal_reachable)
 
     @callback
     def async_add_device_callback(
-        self, resource_type, device=None, force: bool = False
+        self,
+        resource_type: str,
+        device: DeconzAlarmSystem
```

Comment:
Could this be shortened down to `DeconzDevice | None`?

Reply:
Ah I see; I've just tried it with `device: DeconzLight | None = None` (obviously wrong) and mypy doesn't complain.

But I also noticed that the annotation is missing the the possibility of `device` being a `list`, as implied in the line `if device is not None and not isinstance(device, list):` below it.

Reply:
Thanks!

Reply:
Fixed